client/src/fluid/Fluid.ml

```diff
@@ -4162,8 +4162,34 @@ let rec updateKey
     | Keypress {key = K.ShiftEnter; _}, left, _ ->
         let doPipeline (astInfo : ASTInfo.t) : ASTInfo.t =
           let startPos, endPos = FluidUtil.getSelectionRange astInfo.state in
-          let topmostID = getTopmostSelectionID startPos endPos astInfo in
-          let findParent = startPos = endPos in
+          let topmostSelectionID =
+            getTopmostSelectionID startPos endPos astInfo
+          in
+          let topmostID, findParent =
+            if startPos = endPos
+            then
+              let tokenAtLeft =
+                getLeftTokenAt
+                  astInfo.state.newPos
+                  (ASTInfo.activeTokenInfos astInfo)
+              in
+              match tokenAtLeft with
+              | Some current when T.isPipeable current.token ->
+                  (Some (T.tid current.token), false)
+              | _ ->
+                ( match topmostSelectionID with
+                | Some id ->
+                    (Some id, startPos = endPos)
+                | None ->
+                    (Some T.fakeid, startPos = endPos) )
```

Comment:
I'm unclear on why we'd use `Some T.fakeid` instead of `None`? If it is a good idea, can you make topmostID no longer be an option anymore?

Reply:
We shouldn't be using T.fakeid here. Replaced with None.
Taking the option out of topmostID is a bit beyond my OCaml refactoring ability at the moment.

Reply:
If we're keeping none in, then no need to refactor :)